core/system/src/page_allocator_faults.c

```diff
@@ -57,7 +57,10 @@ int page_allocator_get_active_region_for_address(uint32_t address, uint32_t * st
         return UVISOR_ERROR_PAGE_INVALID_PAGE_ORIGIN;
     }
     /* Check that the page actually belongs to this box or box 0. */
-    if ((g_page_owner_table[p] != 0) && (g_page_owner_table[p] != box_id)) {
+    /* We OR together the page owner map for box 0 and the active box. */
+    uint32_t active_page_owner = g_page_owner_map[0][p / 32] | g_page_owner_map[box_id][p / 32];
```

Comment:
@niklas: whenever you set or reset an access bit for box 0, you re-calculate all private box access bits. THat way on a fault you need to check only one bit. For expanding your access windows, the difference between box 0 / box n access is irrelevent, so you get potentially larger access regions with less ACLs